pkg/storage/unified/sql/db/dbimpl/db.go

```diff
@@ -64,3 +104,37 @@ func (tx sqlTx) QueryRowContext(ctx context.Context, query string, args ...any)
 	// and the parameters are passed as arguments."
 	return tx.Tx.QueryRowContext(ctx, query, args...)
 }
+
+// isTransientConnError checks if an error is a transient connection error that should be retried.
+// This includes TCP timeouts, connection refused, and other network-related errors that can occur
+// during Cloud SQL proxy pod churn or similar infrastructure events.
+func isTransientConnError(err error) bool {
```

Comment:
not sure how exhaustive we want this to be. `i/o timeout` is the main one were looking for